code/src/sixsq/nuvla/server/resources/credential.clj

```diff
@@ -173,20 +178,43 @@ passwords) or other services (e.g. TLS credentials for Docker). Creating new
 (defmethod crud/add resource-type
   [{:keys [body] :as request}]
   (let [authn-info (auth/current-authentication request)
+        user-id (auth/current-user-id request)
         desc-attrs (u/select-desc-keys body)
         [create-resp {:keys [id] :as body}]
         (-> body
-            (assoc :resource-type create-type)
-            (update-in [:template] dissoc :subtype)         ;; forces use of template reference
-            (resolve-hrefs authn-info)
-            (update-in [:template] merge desc-attrs)        ;; ensure desc attrs are validated
-            crud/validate
-            :template
-            (tpl->credential request))]
-    (-> request
-        (assoc :id id :body (merge body desc-attrs))
-        add-impl
-        (update-in [:body] merge create-resp))))
+          (assoc :resource-type create-type)
+          (update-in [:template] dissoc :subtype)         ;; forces use of template reference
+          (resolve-hrefs authn-info)
+          (update-in [:template] merge desc-attrs)        ;; ensure desc attrs are validated
+          crud/validate
+          :template
+          (tpl->credential request))
+
+        response (-> request
+                   (assoc :id id :body (merge body desc-attrs))
+                   add-impl
+                   (update-in [:body] merge create-resp))
+
+        id    (:resource-id (:body response))
+
+        acl {:owners   ["group/nuvla-admin"]
+             :view-acl [user-id]}]
+
+    (when (= (:method body) swarm-tpl/method)
```

Comment:
IMO it's better to not combine operations on the resource server-side. We can launch check credential job from the UI after the add. What do you think?

Reply:
what if you create the credential from the API? like the NB credential...

Reply:
It will be checked first time the user will go to deployment modal

Reply:
IMO a) that's not user friendly b) it is not consistent, cause we would be giving users different workflows depending on whether they use the UI or the API c) it doesn't scale, cause if I install 100 NBs, and only then at launch time I have to wait for 100 jobs to run... that's not good d) again, if I'm using the API to also deploy, I'll never have any information about the IS status and swarm-enabled capability

Reply:
Okey, C is a good reason to launch check job on add since tagging of swarm capability is done by side-effect of the check credential.